iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.inl

```diff
@@ -60,11 +61,15 @@ inline constexpr Duration Duration::zero() noexcept
     return Duration{0U, 0U};
 }
 
+// AXIVION Next Construct AutosarC++19_03-A3.9.1 : Use of unsigned long long int in user-defined
+// literals is enforced by the standard
```

Comment:
This I do not get. Why do we not return a `uint64_t`? Or keep it `T` but clamp to zero (relevant only for signed types of course).

For the latter: we might not want to keep a signed type in this case and it is kind of hard to choose a corresponding unsigned type. It is possible with templates like `signed_type_of<T>` (to be defined), and we can even enforce the non-supported types by making it fail to compile for them.

Disregarding that, why not `uint64_t` if it is non-negative to avoid the comment?

Reply:
@MatthiasKillat Ah, good catch, thanks! I had incorrectly treated this case the same as the string literals on lines 524+.